bundles/org.openhab.binding.roborock/src/main/java/org/openhab/binding/roborock/internal/RoborockAccountHandler.java

```diff
@@ -0,0 +1,489 @@
+                    token = loginResponse.data.token;
+                    rriot = loginResponse.data.rriot;
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unsuccessful login: "
+                            + JsonParser.parseString(response).getAsJsonObject().get("msg").getAsString());
+                    return;
+                }
+            } catch (RoborockException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Error " + e.getMessage());
+            }
+        }
+        updateStatus(ThingStatus.ONLINE);
+        mqttConnectTask.submit();
+    }
+
+    private synchronized void teardown(boolean scheduleReconnection) {
+        disconnectMqttClient();
+
+        mqttConnectTask.cancel();
+        initTask.cancel();
+
+        if (scheduleReconnection) {
+            initTask.submit();
+        }
+    }
+
+    private void establishMQTTConnection() {
+        if (token.isEmpty() || rriot.r == null || rriot.r.m.isEmpty() || rriot.k.isEmpty() || rriot.s.isEmpty()
+                || rriot.u.isEmpty()) {
+            logger.debug("token and/or rriot are empty, delay connection to MQTT server");
+            return;
+        }
+
+        try {
+            connectMqttClient();
+            logger.debug("Bridge connected to MQTT");
+            updateStatus(ThingStatus.ONLINE);
+        } catch (MqttException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.comm-error.no-mqtt");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        teardown(false);
+    }
+
+    public void connectMqttClient() throws MqttException {
+        try {
+            URI mqttURL = new URI(rriot.r.m);
+            mqttUser = ProtocolUtils.md5Hex(rriot.u + ':' + rriot.k).substring(2, 10);
+            String mqttPassword = ProtocolUtils.md5Hex(rriot.s + ':' + rriot.k).substring(16);
+
+            String serverURI = "ssl://" + mqttURL.getHost() + ":" + mqttURL.getPort();
+            String clientId = mqttUser;
+
+            mqttClient = new MqttClient(serverURI, clientId, new MemoryPersistence());
+            mqttClient.setCallback(this);
+
+            MqttConnectOptions connOpts = new MqttConnectOptions();
+            connOpts.setCleanSession(false);
+            connOpts.setUserName(mqttUser);
+            connOpts.setPassword(mqttPassword.toCharArray());
+            connOpts.setAutomaticReconnect(true);
+            connOpts.setConnectionTimeout(60);
+            connOpts.setKeepAliveInterval(0);
+
+            logger.debug("Connecting to MQTT broker at {}", serverURI);
+            mqttClient.connect(connOpts);
+            logger.debug("Established MQTT connection.");
+        } catch (URISyntaxException e) {
+            logger.error("Malformed MQTT URL", e);
```

Comment:
This should lead to a CONFIGURATION_ERROR no?

Reply:
well it shouldn't occur at all, given we already have a successful login...

Reply:
fixed.